parlai/core/torch_ranker_agent.py

```diff
@@ -117,23 +151,16 @@ def train_step(self, batch):
         scores = self.score_candidates(batch, cand_vecs)
         loss = self.rank_loss(scores, label_inds)
 
-        # Update metrics
+        # Update loss
         self.metrics['loss'] += loss.item()
         self.metrics['examples'] += batchsize
-        _, ranks = scores.sort(1, descending=True)
-        for b in range(batchsize):
-            rank = (ranks[b] == label_inds[b]).nonzero().item()
-            self.metrics['rank'] += 1 + rank
-
         loss.backward()
         self.update_params()
 
-        # Get predictions but not full rankings for the sake of speed
-        if cand_vecs.dim() == 2:
-            preds = [cands[ordering[0]] for ordering in ranks]
-        elif cand_vecs.dim() == 3:
-            preds = [cands[i][ordering[0]] for i, ordering in enumerate(ranks)]
-        return Output(preds)
+        # Get train predictions
+        if self.opt.get('train_predict', True):
+            return self.get_train_preds(scores)
```

Comment:
in the else case, can we throw a warn_once that some training metrics are omitted for training speed?

Is there a middle ground where we produce partial metrics or something?

Reply:
We still calculate train loss. We just omit accuracy and rank.

Reply:
and also f1 i guess